src/App.js

```diff
@@ -28,16 +37,17 @@ const App = () => {
       <div className="App">
         <Switch>
           <Route exact path="/">
-            <Home />
+            <Home setToken={setToken} />
           </Route>
           <Route exact path="/list">
-            <List />
+            <List token={token} />
```

Comment:
Nice work!!! On other thing you can do is put the redirects here and keep all of the routing together.

Something like... 
```suggestion
                    {token ? <List token={token} /> : <Redirect to="/" /> }
```

Reply:
Keeping the routes all together is a good call. Going to implement that now. Thank you!